sparsecoding/inference/iht.py

```diff
@@ -9,25 +9,23 @@ class IHT(InferenceMethod):
     Infer coefficients for each image in data using elements dictionary.
     Method description can be traced to
     "Iterative Hard Thresholding for Compressed Sensing" (T. Blumensath & M. E. Davies, 2009)
+
+    Parameters
+    ----------
+    sparsity : scalar (1,)
+        Sparsity of the solution. The number of active coefficients will be set
+        to ceil(sparsity * data_dim) at the end of each iterative update.
+    n_iter : scalar (1,) default=100
+        number of iterations to run for an inference method
+    return_all_coefficients : string (1,) default=False
+        returns all coefficients during inference procedure if True
```

Comment:
i can see this being slightly confusing (which dimension does "all" refer to), maybe change wording to something like "returns coefficients for all inference steps"
and also do for all methods with this arg

btw there are some inconsistencies with this arg across inference methods so i opened another issue i can fix separately https://github.com/rctn/sparsecoding/issues/93

Reply:
Thanks for making this issue. I think it's beyond the scope of this PR. Moreover, `return_all_coefficients` for IHT doesn't work as of right now.